odoo/addons/base/models/ir_model.py

```diff
@@ -1415,6 +1415,10 @@ def safe_write(records, fname, value):
                 records.invalidate_cache([fname])
 
         for selection in self:
+            if selection.field_id.model not in self.env:
+                _logger.warning('Selection field "%s" references to non-existing model "%s". You may need to add migration script', selection.field_id.name, selection.field_id.model)
```

Comment:
We're encountering this issue with these steps:

1. create a module that defines a few models, one of them defining a selection field
2. install the module on a database
3. remove the model that defined the selection field from the code
4. trigger an update the module


Simply removing the model will cause this issue.

IMO, the warning doesn't make sense in this situation, so I would just remove it

```suggestion
```

In fact, this case is similar to the one covered a few lines below: https://github.com/odoo/odoo/blob/e34cef712318270cdb3306a2238ac9cbb1bab50c/odoo/addons/base/models/ir_model.py#L1423-L1430

Only that the case where the model itself is deleted was forgotten 😓